src/System.Management.Automation/engine/Utils.cs

```diff
@@ -650,30 +650,31 @@ internal static string GetRegistryConfigurationPath(string shellID)
 #else
             lock (s_cachedGroupPolicySettings)
             {
-                // Return cached information, if we have it
-                Dictionary<string, object> settings;
-                if ((s_cachedGroupPolicySettings.TryGetValue(settingName, out settings)) &&
```

Comment:
A Tuple with settingName, groupPolicyBase, and preferenceOrder should have the needed uniqueness.  But this PR change also works.  It is not clear that Tuple hashing would be significantly faster than string hashing.